src/seedu/addressbook/util/FilePathUtil.java

```diff
@@ -0,0 +1,50 @@
+package seedu.addressbook.util;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * This is a utility class for file path validation.
+ */
+public class FilePathUtil {
+    private static final int EXTENSION_INVALID_INDEX = 0;
+
+    /**
+     * Returns true if the given file path is acceptable.
+     * The file path is acceptable if it has a folder that exists,
+     * has a file name that is acceptable to the operating system eg. does not contain <>:"/|?*,
+     * File name must also contain an extentsion
+     */
+    public static boolean isValidFilePath(String filePath) {
+        if (filePath == null) {
+            return false;
+        }
+        File filePathToValidate = new File(filePath);
+        if (!hasValidFileName(filePathToValidate)) {
+            return false;
+        }
+        return hasValidFolder(filePathToValidate);
+    }
+    
+    /**
+     * Returns true if the file path has a folder that exists
+     */
+    private static boolean hasValidFolder(File filePath) {
+        File folderToValidate = filePath.getAbsoluteFile().getParentFile();
+        return folderToValidate.exists();
+    }
+    
+    /**
+     * Returns true if file path has a valid file name.
+     * File name is valid if it has an extension and no reserved characters <>:"/|?*.
+     */
+    private static boolean hasValidFileName(File filePath) {
+        try {
+            File fileNameToValidate = new File(filePath.getName()).getCanonicalFile();
```

Comment:
Hmm, the [`getCanonicalFile()`](https://docs.oracle.com/javase/7/docs/api/java/io/File.html#getCanonicalPath()) docs says that it basically performs path normalization (e.g. resolving symlinks and redundant `..` components), which may be overkill for what this method wants to do.

I understand that it is needed to check for invalid characters (which is platform-dependent), and it is indeed annoying that the `File` API does not bother to validate whether the path it contains is valid or not.

Thankfully, it seems Java has a newer [`java.nio.file`](https://docs.oracle.com/javase/7/docs/api/java/nio/file/package-summary.html) API. In particular, it has a [`Paths::get()`](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Paths.html#get(java.lang.String,%20java.lang.String...)), which actually does perform the platform-dependent path validation. Perhaps this newer and less-clunky API should be used instead?